sympy/polys/euclidtools.py

```diff
@@ -1248,6 +1247,386 @@ def dup_zz_heu_gcd(f, g, K):
+
+    Returns:
+        R: The list of polynomials representing the subresultant sequence.
+
+    Raises:
+        ValueError: If an unexpected condition occurs during the computation.
+
+    Notes:
+        - The subresultant sequence is a sequence of polynomials obtained by performing polynomial
+          divisions and subtractions in a specific manner.
+        - The subresultant sequence is used in various polynomial algorithms, such as polynomial
+          greatest common divisor (GCD) computation.
+        - The subresultant sequence is computed recursively until a certain condition is met.
+        - The algorithm used in this function is based on the efficient sparse polynomial
+          representation and manipulation techniques.
+
+    Examples:
+
+
+
+    """
+
+    n = f.degree(x)
+    m = g.degree(x)
+
+    if n < m:
+        f, g = g, f
+        n, m = m, n
+
+    if f == 0:
+        return [0, 0]
+
+    if g == 0:
+        return [f, 1]
+
+    # Initialize the subresultant sequence
+    R = [f, g]
+
+    d = n - m
+    b = (-1) ** (d + 1)
+
+    # Compute the initial premultiplication factor for the next polynomial
+    h = compute_prem_sparse(f, g, x)
+    h = h * b
+
+    # Compute the leading coefficient of g
+    lc = get_sparse_coefficient(g, x, m)
+
+    c = lc ** d
+
+    S = [1, c]
+
+    c = -c
+
+    # Iterate until the final polynomial is obtained
+    while h:
+        k = h.degree(x)
+
+        R.append(h)
+        f, g, m, d = g, h, k, m - k
+
+        # Compute the premultiplication factor for the next polynomial
+        b = -lc * c ** d
+        h = compute_prem_sparse(f, g, x)
+        h = h.exquo(b)
+
+        lc = get_sparse_coefficient(g, x, k)
+
+        # Update the constant factor
+        if d > 1:
+            p = (-lc) ** d
+            q = c ** (d - 1)
+            c = p.exquo(q)
+        else:
+            c = -lc
+
+        S.append(-c)
+
+    return R
+
+
+def compute_primitive_sparse(p, x):
+    """
+    Computes the primitive part and content of a polynomial `p` with respect to the variable `x`.
+    """
+
+    coefficients = get_sparse_coefficients(p, {x})
+    content = compute_gcd_sparse(coefficients)
+    primitive = p.exquo(content)
+    return content, primitive
+
+
+def find_main_variable_sparse(p):
+    """
+    Finds the main variable of a polynomial in sparse representation.
+    """
+    symbols = compute_free_symbols_sparse(p)
+    return min(symbols)
+
+
+def compute_gcd_prs_sparse(p1, p2):
+    """
+    Computes the greatest common divisor (GCD) of two polynomials using the Polynomial Resultant Sequences (PRS) method.
+    """
+    x = find_main_variable_sparse(p1)
+
+    c1, pp1 = compute_primitive_sparse(p1, x)
+    c2, pp2 = compute_primitive_sparse(p2, x)
+
+    h = compute_subresultants_sparse(pp1, pp2, x)[-1]
+    c = compute_gcd_sparse([c1, c2])
+
+    ring = p1.ring.to_domain()
+    if ring.is_negative(get_sparse_coefficient(h, x, h.degree(x))):
+        h = -h
```

Comment:
This should probably use `canonical_unit` rather than `is_negative` to handle the case of Gaussian integers like `ZZ_I` or `QQ_I`.